src/index.ts

```diff
@@ -4,64 +4,7 @@
 // *** WITH THE PHALA TEAM AT https://discord.gg/5HfmWQNX THANK YOU             ***
 // *** FOR DOCS ON HOW TO CUSTOMIZE YOUR PC 2.0 https://bit.ly/customize-pc-2-0 ***
 import "@phala/pink-env";
-import { Coders } from "@phala/ethers";
-
-type HexString = `0x${string}`
-
-// ETH ABI Coders available
-/*
-// Basic Types
-// Encode uint
-const uintCoder = new Coders.NumberCoder(32, false, "uint256");
-// Encode Bytes
-const bytesCoder = new Coders.BytesCoder("bytes");
-// Encode String
-const stringCoder = new Coders.StringCoder("string");
-// Encode Address
-const addressCoder = new Coders.AddressCoder("address");
-
-// ARRAYS
-//
-// ***NOTE***
-// IF YOU DEFINE AN TYPED ARRAY FOR ENCODING, YOU MUST ALSO DEFINE THE SIZE WHEN DECODING THE ACTION REPLY IN YOUR
-// SOLIDITY SMART CONTRACT.
-// EXAMPLE for an array of string with a length of 10
-//
-// index.ts
-const stringCoder = new Coders.StringCoder("string");
-const stringArrayCoder = new Coders.ArrayCoder(stringCoder, 10, "string[]");
-function encodeReply(reply: [number, number, string[]]): HexString {
-  return Coders.encode([uintCoder, uintCoder, stringArrayCoder], reply) as HexString;
-}
-
-const stringArray = string[10];
-
-export default function main(request: HexString, secrets: string): HexString {
-  return encodeReply([0, 1, stringArray]);
-}
-// OracleConsumerContract.sol
-function _onMessageReceived(bytes calldata action) internal override {
-    (uint respType, uint id, string[10] memory data) = abi.decode(
-        action,
-        (uint, uint, string[10])
-    );
-}
-// Encode Array of addresses with a length of 10
-const stringArrayCoder = new Coders.ArrayCoder(stringCoder, 10, "string");
-// Encode Array of addresses with a length of 10
-const addressArrayCoder = new Coders.ArrayCoder(addressCoder, 10, "address");
-// Encode Array of bytes with a length of 10
-const bytesArrayCoder = new Coders.ArrayCoder(bytesCoder, 10, "bytes");
-// Encode Array of uint with a length of 10
-const uintArrayCoder = new Coders.ArrayCoder(uintCoder, 10, "uint256");
-*/
-
-const uintCoder = new Coders.NumberCoder(32, false, "uint256");
-const bytesCoder = new Coders.BytesCoder("bytes");
-
-function encodeReply(reply: [number, number, number]): HexString {
-  return Coders.encode([uintCoder, uintCoder, uintCoder], reply) as HexString;
-}
+import {encodeReply, decodeRequest, HexString, encodeReplyAbiParams, decodeRequestAbiParams} from "./viem/coder";
```

Comment:
I think it's better to write it in one file. After the playground feature is live, developers can simply copy and paste to run.

Reply:
Moved back into `index.ts` now

Reply:
Is it possible to reduce the starter kit line of code to ~50ish?

Reply:
@HashWarlock @h4x3rotab Do you think this template should only echo a response without an HTTP request? It's not what we need to do in this PR, but I believe the starter kit would be better as an empty template.

Reply:
If we create a parser that can parse the graphql queries and another parser to parse http request results then we can pass the graphql query and the parsing details of the results for each query in the `secrets` which can reduce the size. So the template, we can have `secrets` passed in like:
```
{
  "apiUrl": "https://api-v2.lens.dev",
  "httpRequests": [
    {
      "query": "query Profile { profile(request: { forProfileId: '${requestStr}' }) {
          stats {
              followers
              following
              comments
              countOpenActions
              posts
              quotes
              mirrors
              publications
              reacted
              reactions
          }
        }
      }",
     "parseInfo": {
        "path": "data.profile.stats.posts",
        "defaultValue": 0,
      }
}
```

Reply:
Please do not move the httpRequest from codes to configurable secrets. It is not a good decision.

Reply:
A few things to consider:
1. Can the json path be hard coded in the code? Why do you make it configurable?
2. Is it possible to simplify `encodeReply`?

Reply:
1. I made it configurable to reduce code size, but it doesn't make sense to do this now.
2. Added change to Simplify `encodeReply`